File: Baekjoon/구현/25304.py
# https://www.acmicpc.net/problem/25304

import sys
input = sys.stdin.readline

x = int(input())
n = int(input())
total = 0

for _ in range(n):
    a, b = map(int, input().split())
    total += a * b

if total == x:
    print('Yes')
else:
    print('No')

File: Baekjoon/구현/25304.cpp
// https://www.acmicpc.net/problem/25304

#include <iostream>
using namespace std;

int main() {
    int x, n, total;
    total = 0;
    cin >> x >> n;

    for (int i = 0; i < n; i++) {
        int a, b;
        cin >> a >> b;
        total += a * b;
    }

    if (total == x) cout << "Yes" << "\n";
    else cout << "No" << "\n";

    return 0;
}